diff --git a/tests/onebox.go b/tests/onebox.go
index fd039a44d..e437d6d7d 100644
--- a/tests/onebox.go
+++ b/tests/onebox.go
@@ -42,6 +42,8 @@ import (
 	"go.temporal.io/api/operatorservice/v1"
 	"go.temporal.io/api/workflowservice/v1"
 
+	ydbpersistence "github.com/yandex/temporal-over-ydb/persistence"
+	ydbvisibility "github.com/yandex/temporal-over-ydb/visibility"
 	otelsdktrace "go.opentelemetry.io/otel/sdk/trace"
 
 	"go.temporal.io/server/api/adminservice/v1"
@@ -432,8 +434,12 @@ func (c *temporalImpl) startFrontend(hosts map[primitives.ServiceName][]string,
 		// fx.Provide(func() searchattribute.Mapper { return NewSearchAttributeTestMapper() }),
 		fx.Provide(func() resolver.ServiceResolver { return resolver.NewNoopResolver() }),
 		fx.Provide(persistenceClient.FactoryProvider),
-		fx.Provide(func() persistenceClient.AbstractDataStoreFactory { return nil }),
-		fx.Provide(func() visibility.VisibilityStoreFactory { return nil }),
+		fx.Provide(func() persistenceClient.AbstractDataStoreFactory {
+			return ydbpersistence.NewYDBAbstractDataStoreFactory()
+		}),
+		fx.Provide(func() visibility.VisibilityStoreFactory {
+			return ydbvisibility.NewYDBVisibilityStoreFactory()
+		}),
 		fx.Provide(func() dynamicconfig.Client { return c.dcClient }),
 		fx.Provide(func() log.Logger { return c.logger }),
 		fx.Provide(resource.DefaultSnTaggedLoggerProvider),
@@ -523,8 +529,12 @@ func (c *temporalImpl) startHistory(
 			// fx.Provide(func() searchattribute.Mapper { return NewSearchAttributeTestMapper() }),
 			fx.Provide(func() resolver.ServiceResolver { return resolver.NewNoopResolver() }),
 			fx.Provide(persistenceClient.FactoryProvider),
-			fx.Provide(func() persistenceClient.AbstractDataStoreFactory { return nil }),
-			fx.Provide(func() visibility.VisibilityStoreFactory { return nil }),
+			fx.Provide(func() persistenceClient.AbstractDataStoreFactory {
+				return ydbpersistence.NewYDBAbstractDataStoreFactory()
+			}),
+			fx.Provide(func() visibility.VisibilityStoreFactory {
+				return ydbvisibility.NewYDBVisibilityStoreFactory()
+			}),
 			fx.Provide(func() dynamicconfig.Client { return c.dcClient }),
 			fx.Provide(func() log.Logger { return c.logger }),
 			fx.Provide(resource.DefaultSnTaggedLoggerProvider),
@@ -617,8 +627,12 @@ func (c *temporalImpl) startMatching(hosts map[primitives.ServiceName][]string,
 		fx.Provide(func() searchattribute.Mapper { return nil }),
 		fx.Provide(func() resolver.ServiceResolver { return resolver.NewNoopResolver() }),
 		fx.Provide(persistenceClient.FactoryProvider),
-		fx.Provide(func() persistenceClient.AbstractDataStoreFactory { return nil }),
-		fx.Provide(func() visibility.VisibilityStoreFactory { return nil }),
+		fx.Provide(func() persistenceClient.AbstractDataStoreFactory {
+			return ydbpersistence.NewYDBAbstractDataStoreFactory()
+		}),
+		fx.Provide(func() visibility.VisibilityStoreFactory {
+			return ydbvisibility.NewYDBVisibilityStoreFactory()
+		}),
 		fx.Provide(func() dynamicconfig.Client { return c.dcClient }),
 		fx.Provide(func() *esclient.Config { return c.esConfig }),
 		fx.Provide(func() esclient.Client { return c.esClient }),
@@ -713,8 +727,12 @@ func (c *temporalImpl) startWorker(hosts map[primitives.ServiceName][]string, st
 		fx.Provide(func() searchattribute.Mapper { return nil }),
 		fx.Provide(func() resolver.ServiceResolver { return resolver.NewNoopResolver() }),
 		fx.Provide(persistenceClient.FactoryProvider),
-		fx.Provide(func() persistenceClient.AbstractDataStoreFactory { return nil }),
-		fx.Provide(func() visibility.VisibilityStoreFactory { return nil }),
+		fx.Provide(func() persistenceClient.AbstractDataStoreFactory {
+			return ydbpersistence.NewYDBAbstractDataStoreFactory()
+		}),
+		fx.Provide(func() visibility.VisibilityStoreFactory {
+			return ydbvisibility.NewYDBVisibilityStoreFactory()
+		}),
 		fx.Provide(func() dynamicconfig.Client { return c.dcClient }),
 		fx.Provide(func() log.Logger { return c.logger }),
 		fx.Provide(resource.DefaultSnTaggedLoggerProvider),
diff --git a/tests/test_cluster.go b/tests/test_cluster.go
index 20e885156..8484f0621 100644
--- a/tests/test_cluster.go
+++ b/tests/test_cluster.go
@@ -25,6 +25,7 @@
 package tests
 
 import (
+	temporalypersistencetests "github.com/yandex/temporal-over-ydb/tests/persistencetests"
 	"context"
 	"crypto/tls"
 	"crypto/x509"
@@ -165,7 +166,7 @@ func NewCluster(options *TestClusterConfig, logger log.Logger) (*TestCluster, er
 		options.Persistence.FaultInjection.Rate = TestFlags.PersistenceFaultInjectionRate
 	}
 
-	testBase := persistencetests.NewTestBase(&options.Persistence)
+	testBase := temporalypersistencetests.NewTestBaseWithYDB(&options.Persistence)
 	testBase.Setup(clusterMetadataConfig)
 	archiverBase := newArchiverBase(options.EnableArchival, logger)
 
